s3stream/src/main/java/com/automq/stream/s3/wal/impl/object/RecordAccumulator.java

```diff
@@ -297,6 +315,40 @@ public long append(long recordSize, Function<Long, ByteBuf> recordSupplier,
         }
     }
 
+    public CompletableFuture<Void> reset() {
+        checkStatus();
+
+        if (objectMap.isEmpty() && previousObjectMap.isEmpty()) {
+            return CompletableFuture.completedFuture(null);
+        }
+
+        long startTime = time.nanoseconds();
+        AtomicLong deletedObjectSize = new AtomicLong();
+        List<ObjectStorage.ObjectPath> deleteObjectList = new ArrayList<>();
+
+        previousObjectMap.forEach((k, v) -> {
```

Comment:
remove in foreach is undefined

Reply:
In the implementation of this map, removing an element using an iterator is no different from using a foreach loop.

<img width="857" alt="image" src="https://github.com/user-attachments/assets/a32d26bb-f758-495f-ad3c-c08624e70bb5">